funcbench/main.go

```diff
@@ -0,0 +1,281 @@
+					data, err = exec.Command("rm", path).CombinedOutput()
+					if err != nil {
+						log.Fatalln("Error: ", err, "Command out: ", string(data))
+					}
+					return nil
+				}
+				log.Fatalln(err)
+			}
+			return nil
+		})
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Benchmark the comparing branch.
+		if race == "-no-race" {
+			cmd = exec.Command("go", "test", "-bench", fmt.Sprintf("^%s$", issueComment), "-benchmem", "-v", "./...")
+		} else {
+			cmd = exec.Command("go", "test", "-bench", fmt.Sprintf("^%s$", issueComment), "-benchmem", "-race", "-v", "./...")
+		}
+		data, err = cmd.CombinedOutput()
+		log.Println(string(data))
+		if err != nil || cmd.ProcessState.ExitCode() != 0 {
+			if err := postComment(clt, owner, repo, prNumber, fmt.Sprintf("Go test on branch %s failed. %s", branch, logLink)); err != nil {
+				log.Fatalln(err)
+			}
+			log.Fatalln(err)
+		}
+		err = ioutil.WriteFile(oldPath, data, 0644)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Compare the benchmarks.
+		data, err = exec.Command(strings.Join([]string{os.Getenv("GOPATH"), "/bin/benchcmp"}, ""), oldPath, newPath).CombinedOutput()
+		if err != nil {
+			log.Println(string(data))
+			if err := postComment(clt, owner, repo, prNumber, fmt.Sprintf("Error: `benchcmp` failed. %s", logLink)); err != nil {
+				log.Fatalln(err)
+			}
+			log.Fatalln(err)
+		}
+		if strings.Count(string(data), "\n") < 2 {
+			if err := postComment(clt, owner, repo, prNumber, fmt.Sprintf("Error: `go test` did not match any benchmarked functions. %s", logLink)); err != nil {
+				log.Fatalln(err)
+			}
+		}
+
+		// Create and post the final comment.
+		tableContent := strings.Split(string(data), "\n")
+		for i := 0; i <= len(tableContent)-1; i++ {
+			e := tableContent[i]
+			switch {
+			case e == "":
+
+			case strings.Contains(e, "old ns/op"):
+				e = "| Benchmark | Old ns/op | New ns/op | Delta |"
+				tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+			case strings.Contains(e, "old MB/s"):
+				e = "| Benchmark | Old MB/s | New MB/s | Speedup |"
+				tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+			case strings.Contains(e, "old allocs"):
+				e = "| Benchmark | Old allocs | New allocs | Delta |"
+				tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+			case strings.Contains(e, "old bytes"):
+				e = "| Benchmark | Old bytes | New bytes | Delta |"
+				tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+			default:
+				// Replace spaces with "|".
+				e = strings.Join(strings.Fields(e), "|")
+			}
+			tableContent[i] = e
+		}
+
+		if err := postComment(clt, owner, repo, prNumber, strings.Join(tableContent, "\n")); err != nil {
+			log.Fatalln(err)
+		}
+
+	default:
+		log.Fatalln("simpleargs only supports issue_comment event")
```

Comment:
```suggestion
		log.Fatalln("only issue_comment event are supported")
```

Reply:
This one is copied from the comment monitor, should I fix it there as well?

Reply:
@geekodour 	is updating it now so can do it as part of his PR